compiler/z/codegen/OMRInstruction.cpp

```diff
@@ -653,28 +653,45 @@ bool isMatchingStoreRestore(TR::Instruction *cursorLoad, TR::Instruction *cursor
    return false;
    }
 
-uint32_t getRegMaskFromRange(TR::Instruction * inst)
+uint64_t getRegMaskFromRange(TR::Instruction * inst, TR::CodeGenerator *cg)
    {
-   uint32_t regs = 0;
+   uint64_t regs = 0;
+   uint32_t maxRegNum, minRegNum;
+   TR::Register *lowReg, *highReg;
+
+   if (inst->getOpCodeValue() == TR::InstOpCode::VLM)
+      {
+      lowReg = toS390VRSInstruction(inst)->getFirstRegister();
+      highReg = toS390VRSInstruction(inst)->getSecondRegister();
+      minRegNum = TR::RealRegister::FirstAssignableVRF;
+      maxRegNum = TR::RealRegister::LastAssignableVRF;
+      }
+   else
+      {
+      lowReg = toS390RSInstruction(inst)->getFirstRegister();
+      highReg = toS390RSInstruction(inst)->getSecondRegister();
+      minRegNum = TR::RealRegister::FirstGPR;
+      maxRegNum = TR::RealRegister::LastAssignableGPR;
+      }
 
-   TR::Register *lowReg = toS390RSInstruction(inst)->getFirstRegister();
-   TR::Register *highReg = toS390RSInstruction(inst)->getSecondRegister();
    if (inst->getRegisterOperand(1)->getRegisterPair())
       highReg = inst->getRegisterOperand(1)->getRegisterPair()->getHighOrder();
 
-   uint32_t lowRegNum = ANYREGINDEX(toRealRegister(lowReg)->getRegisterNumber());
-   uint32_t highRegNum = ANYREGINDEX(toRealRegister(highReg)->getRegisterNumber());
+   uint32_t lowRegNum = toRealRegister(lowReg)->getRegisterNumber();
```

Comment:
What was the purpose of using Macro originally if it was OK without it?

Reply:
`ANYREGINDEX` moves a GPR regnum to the range 0-15, so it could be used with a shift to generate the mask [1]. The problem is that the macro also moves a vector regnum to the range 0-31, so vector registers would be represented in the wrong bits of the mask. I removed the macro and used `getRealRegisterMask` [2] instead of a shift to avoid this problem. This is similar to how the mask is generated in `genBitMapOfAssignableGPRs` [3].

[1] https://github.com/eclipse/omr/blob/fe4c3b9b51f6260af1489994754d970af5f3f9eb/compiler/z/codegen/OMRInstruction.cpp#L671
[2] https://github.com/Spencer-Comin/omr/blob/d4d50377ffc1e10696be68032f9eb683eb2222cd/compiler/z/codegen/OMRInstruction.cpp#L690
[3] https://github.com/eclipse/omr/blob/fe4c3b9b51f6260af1489994754d970af5f3f9eb/compiler/z/codegen/OMRMachine.cpp#L486

Reply:
Thanks a lot @Spencer-Comin  for the clarification. I can see how messy it could have become if you have kept the use of `ANYREGINDEX`. Using the code-generator object to get the mask from machine object is better than calculating that manually.